src/codeflare_sdk/cluster/auth.py

```diff
@@ -58,65 +103,176 @@ def __init__(self, token: str = None, server: str = None, skip_tls: bool = False
+                                    self.username, cluster_name, self.token
+                                )
+                            )
+                            continue
+                        write_file.write(line)
+
+                response = "Updated config file at %s/.kube/config" % home
             else:
-                return error_msg
-        return response.out()
+                # Create a new config file from the config template and store it in HOME/.kube
+                file = open("%s/.kube/config" % home, "w")
+                file.write(
+                    template.render(
+                        security=security,
+                        server=server,
+                        cluster=cluster_name,
+                        context_name="default/{}/{}".format(
+                            cluster_name, self.username
+                        ),
+                        current_context="default/{}/{}".format(
+                            cluster_name, self.username
+                        ),
+                        username="{}/{}".format(self.username, cluster_name),
+                        token=self.token,
+                    )
+                )
+                response = (
+                    "Logged in and created new config file at %s/.kube/config" % home
+                )
+        except:
+            response = "Error logging in. Have you inputted correct credentials?"
+        return response
 
     def logout(self) -> str:
         """
-        This function is used to logout of an OpenShift cluster.
+        This function is used to logout of a Kubernetes cluster.
         """
-        args = [f"--token={self.token}", f"--server={self.server}"]
-        response = oc.invoke("logout", args)
-        return response.out()
+        home = os.path.expanduser("~")
+        file = open(r"%s/.kube/config" % home, "r")
+        lines = file.readlines()
+        line_count = 0
+        for line in lines:
+            if (
+                "- name: {}/{}".format(self.username, self.server[8:].replace(".", "-"))
+                not in line.strip()
+            ):
+                line_count = line_count + 1
+            else:
+                break
+        # The name, user and token are removed from the config file
+        with open(r"%s/.kube/config" % home, "w") as file:
+            for number, line in enumerate(lines):
+                if number not in [line_count, line_count + 1, line_count + 2]:
+                    file.write(line)
+        print("logged out of user %s" % self.username)
 
 
-class PasswordUserAuthentication(Authentication):
+class KubeConfigFileAuthentication(KubeConfiguration):
     """
-    `PasswordUserAuthentication` is a subclass of `Authentication`. It can be used to authenticate to an OpenShift
-    cluster when the user has a username and password.
+    An abstract class that defines the necessary methods for passing a user's own Kubernetes config file.
```

Comment:
```suggestion
    A class that defines the necessary methods for passing a user's own Kubernetes config file.
```
I don't think this is an abstract class as its a subclass of KubeConfiguraiton.